argo/src/Argo/DefaultMain.hs

```diff
@@ -21,14 +25,36 @@ import System.Exit (die)
 import Argo
 import Argo.Socket
 
+data UserOptions stdIOOpts socketOpts httpOpts
+  = StdIOOpts stdIOOpts
+  | SocketOpts socketOpts
+  | HttpOpts httpOpts
 
-defaultMain :: String -> App s -> IO ()
-defaultMain str app =
+userOptions :: UserOptions a a a -> a
+userOptions (StdIOOpts opts) = opts
+userOptions (SocketOpts opts) = opts
+userOptions (HttpOpts opts) = opts
+
+customMain ::
```

Comment:
Now that there is both a custom and default main, it might be nice to have some simple Haddock to help newcommers and/or forgetful folks quickly grok the intended differences.

Reply:
Done! Good call.